tests/kohana/LogTest.php

```diff
@@ -61,51 +61,360 @@ public function test_attach_attaches_log_writer_and_returns_this()
 		$this->assertSame($logger, $logger->attach($writer));
 
 		$this->assertAttributeSame(
-			array(spl_object_hash($writer) => array('object' => $writer, 'levels' => array())),
+			array(spl_object_hash($writer) => $writer),
 			'_writers',
 			$logger
 		);
 	}
 
 	/**
-	 * Test that attaching a log writer using a min/max level adds it to the array of log writers
-	 *
-	 * @TODO Is this test too specific?
+	 * When we call detach() we expect the specified log writer to be removed
 	 *
 	 * @test
-	 * @covers Log::attach
+	 * @covers Log::detach
 	 */
-	public function test_attach_attaches_log_writer_min_max_and_returns_this()
+	public function test_detach_removes_log_writer_and_returns_this()
 	{
 		$logger = new Log;
 		$writer = $this->getMockForAbstractClass('Log_Writer');
 
-		$this->assertSame($logger, $logger->attach($writer, Log::NOTICE, Log::CRITICAL));
+		$logger->attach($writer);
 
-		$this->assertAttributeSame(
-			array(spl_object_hash($writer) => array('object' => $writer, 'levels' => array(Log::CRITICAL, Log::ERROR, Log::WARNING, Log::NOTICE))),
-			'_writers',
-			$logger
-		);
+		$this->assertSame($logger, $logger->detach($writer));
+
+		$this->assertAttributeSame(array(), '_writers', $logger);
 	}
 
 	/**
-	 * When we call detach() we expect the specified log writer to be removed
+	 * Data provider for test_log_message
+	 *
+	 * @return array
+	 */
+	public function provider_log_message()
+	{
+		// a sample exception used in data sets
+		$exception = new Exception('dummy exception text');
+		return [
+			// data set #0
+			[
+				\Psr\Log\LogLevel::DEBUG, // level
+				'dummy message', // log text
+				[], // context
+				// expected raw log message array
+				[
+					'time' => time(), // to be reset within the test for HHVM
```

Comment:
Maybe don't set the time at all in the data provider and just add it in the test message, to make it most obvious?